Lib/test/test_ssl.py

```diff
@@ -4346,8 +4346,9 @@ def test_client_sigalgs_mismatch(self):
         client_context.set_client_sigalgs("rsa_pss_rsae_sha256")
         server_context.set_client_sigalgs("rsa_pss_rsae_sha384")
 
-        # Some systems return ConnectionResetError on handshake failures
-        with self.assertRaises((ssl.SSLError, ConnectionResetError)):
+        # gh-139504: Some systems return ConnectionResetError
+        # or BrokenPipeError on handshake failures
+        with self.assertRaises((ssl.SSLError, ConnectionResetError, BrokenPipeError)):
```

Comment:
```suggestion
        with self.assertRaises((
            ssl.SSLError,
            # On handshake failures, some systems raise a ConnectionResetError.
            ConnectionResetError,
            # On handshake failures, macOS may raise a BrokenPipeError.
            # See https://github.com/python/cpython/issues/139504.
            BrokenPipeError,
        )):
```

I'd like to be able to fix this BrokenPipeError if possible but I don't know how for now. So I'd prefer making it explicit that ssl.SSLError is the expected exception while the others are for flakyiness reasons.

Reply:
I think we're exactly at 80 chars (I just want to have the same comment structure, so just fit them under 80 chars)